lib/Sema/CSSolver.cpp

```diff
@@ -1751,19 +1752,19 @@ void ConstraintSystem::shrink(Expr *expr) {
       if (auto array = dyn_cast<ArraySliceType>(base)) {
         auto elementType = array->getBaseType();
         // If base type is invalid let's return error type.
```

Comment:
Why would it be an invariant? An array slice type can end up with an error type inside of it